### PR TITLE
Dm 1928

### PR DESCRIPTION
**Pull request by JoshingYou1**
### JIRA issue link
https://agile6.atlassian.net/browse/DM-1928

## Description - what does this code do?
Allows admin users to add categories to their practice.

## Testing done - how did you test it/steps on how can another person can test it 
1. Log in as an admin.
2. Browse to the practice editor introduction page of a practice of your choosing.
3. Add/remove categories as you see fit.
4. Check the `Other` option and add/remove custom categories.
5. Visit the same practice's show page and make sure all categories that **_are not Other or custom_** are displayed below the thumbnail image at the top right.
6. Use developer tools and choose a viewport width smaller than `1024px`.
7. Make sure the category tag list is at the bottom of the page, below comments, instead of at the top.
8. Click on one of the category tags.
9. Make sure you are redirected to the search results page and that the correct results are displayed.

**BONUS 1**
1. In the code, visit `app/helpers/practices_helper.rb` and comment out `.where.not(is_other: true, name: 'Other')` from line `108`.
2. Go back to the editor and make sure you have at least 11 total categories assigned, including the `Other` option and all custom categories.
2. Visit the practice's show page and make sure the `SEE MORE` button is now visible and confirm interactivity.

**BONUS 2**
1. Check all of the above on IE :blush:

## Screenshots, Gifs, Videos from application (if applicable)
Screen Shot 2020-08-14 at 12 37 56 PM
Screen Shot 2020-08-14 at 12 38 59 PM
Screen Shot 2020-08-14 at 12 42 06 PM
Screen Shot 2020-08-14 at 12 41 04 PM
Screen Shot 2020-08-14 at 12 41 26 PM

## Link to mock-ups/mock ups (image file if you have it) (if applicable)
https://www.figma.com/file/FogT5mBUVl8rE7OgWDQWeJ/VADM?node-id=276%3A0

## Acceptance criteria
- [Build Categories (Done)

Add ability to add Categories in active Admin (Done)

Categories can have “related terms” (Done) 

List them out below Campaign on practice page

Make Categories searchable (Done)

When Category is clicked on, show search results for that term

Build PE where user can select multiple Categories 

Only let them select from categories that we currently have ]

## Definition of done
- [X] Unit tests written (if applicable)
- [X] e2e/accessibility tests written (if applicable)
- [X] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [X] A link has been provided to the originating JIRA issue
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: app/controllers/practice_origin_facilities.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class PracticeOriginFacilities < ApplicationController
```
Comment (ahay-agile6) [inline app/controllers/practice_origin_facilities.rb:1]: delete this file

File: app/services/save_practice_service.rb [hunk 2]
```diff
@@ -16,6 +16,7 @@ def initialize(params)
         crop_main_display_image: 'error cropping practice thumbnail',
         update_initiating_facility: 'error updating initiating facility',
         update_practice_awards: 'error updating practice awards',
+        # update_category_practices: 'error updating practice categories'
```
Comment (ahay-agile6) [inline app/services/save_practice_service.rb:19]: comment this back in

File: app/services/save_practice_service.rb [hunk 3]
```diff
@@ -31,20 +32,22 @@ def save_practice
       rescue_method(:crop_main_display_image)
       rescue_method(:update_initiating_facility)
       rescue_method(:update_practice_awards)
-
+      # rescue_method(:update_category_practices)
```
Comment (ahay-agile6) [inline app/services/save_practice_service.rb:35]: comment this back in and delete line 41

File: app/services/save_practice_service.rb [hunk 4]
```diff
@@ -162,33 +219,13 @@ def update_practice_awards
     if practice_award_params
       practice_awards_to_create = practice_award_params.values.map { |param| param[:name] }
       practice_awards_to_create.each { |award| @practice.practice_awards.find_or_create_by(name: award) }
-      # get practice awards that are in the provided list
```
Comment (ahay-agile6) [inline app/services/save_practice_service.rb:165]: did we remove this code?
if so.... put it back! i think, lol :laughing:

File: app/views/practices/form/introduction.html.erb [hunk 5]
```diff
@@ -445,6 +445,120 @@
...
+                    <%= f.fields_for :categories, wrapper: false do |cat| %>
+                      <%
+                        other_category_id = cat.object.id || "new_categories"
+                        show_other_category_form = other_category_ids.include?(other_category_id) || other_category_id == "new_categories"
+                      %>
+
+                      <% if show_other_category_form %>
+                        <% if cat.object.id %>
+                          <%= javascript_tag 'data-turbolinks-track': 'reload' do %>
+                            $(document).arrive('<%= "#practice_categories_attributes_#{cat.index}_name" %>',
```
Comment (ahay-agile6) [inline app/views/practices/form/introduction.html.erb:499]: you'll have to manually format this

File: spec/features/practice_editor/contact_spec.rb [hunk 6]
```diff
@@ -110,5 +110,49 @@ def fill_in_contact_fields
             expect(page).to have_field('Name:', with: nil)
             expect(page).to have_field('Role:', with: nil)
         end
+
+        it 'should allow the user to crop multiple contacts -- new and existing' do
```
Comment (ahay-agile6) [inline spec/features/practice_editor/contact_spec.rb:114]: is this why the tests are failing?

Comment (JoshingYou1) [reply spec/features/practice_editor/contact_spec.rb:114]: Yeah, that's the one. The container is getting a `visibility: hidden` attribute, so it errors out.

File: app/services/save_practice_service.rb (hunk 2 above)
Comment (JoshingYou1) [reply app/services/save_practice_service.rb:19]: Done!

File: app/controllers/practice_origin_facilities.rb (hunk 1 above)
Comment (JoshingYou1) [reply app/controllers/practice_origin_facilities.rb:1]: Done!

File: app/services/save_practice_service.rb (hunk 3 above)
Comment (JoshingYou1) [reply app/services/save_practice_service.rb:35]: Done!

**Pull request merged**